components/sceneutil/lightmanager.cpp

```diff
@@ -220,7 +214,8 @@ namespace SceneUtil
         // possible optimization: return a StateSet containing all requested lights plus some extra lights (if a suitable one exists)
         size_t hash = 0;
         for (unsigned int i=0; i<lightList.size();++i)
-            boost::hash_combine(hash, lightList[i]->mLightSource->getId());
+            hash = hash ^ (lightList[i]->mLightSource->getId() << 1); // or use boost::hash_combine
```

Comment:
It was faster doesn't attempt to prevent  the low-end and high-end bits from colliding.

We can just do what boost does, use the golden ratio to prevent (slow down) this collision.

```
template <class T>
inline void hash_combine(std::size_t& seed, const T& v)
{
    std::hash<T> hasher;
    seed ^= hasher(v) + 0x9e3779b9 + (seed<<6) + (seed>>2);
}
```

Do we want good enough or do we want 'the best' but slower.